test/mapwindow.jl

```diff
@@ -91,5 +91,22 @@ using ImageFiltering, Base.Test
                                          2 3 3 4 4;
                                          1 3 3 3 2]
     end
+
+    function groundtruth(f, A, window::Tuple, border, imginds)
+        mapwindow(f,A,window,border)[imginds...]
+    end
+    for (f,img, window, imginds) ∈ [
+            (mean, randn(10), (1,), (1:2:10,)),
+            (median!, randn(10), (-1:1,), (1:2:8,)),
+            (mean, randn(10), (-1:1,), (1:2:8,)),
+            (mean, randn(10,5), (-1:1,0:0), (1:2:8,1:3)),
+            (mean, randn(10,5), (-1:1,0:0), (Base.OneTo(2),1:3)),
+        ]
+        border = "replicate"
+        @test groundtruth(f,img,window,border,imginds) == @inferred mapwindow(f,img,window,border,imginds)
+    end
+    @test mapwindow(mean, randn(10), (3,), "replicate", 2:2:7) isa Array
+    @test mapwindow(mean, randn(10), (3,), "replicate", 2:7) isa OffsetArray
+    @test mapwindow(mean, randn(10), (3,)) isa Array
```

Comment:
Sorry I didn't notice this earlier, but over time I've gotten more skeptical about testing the output type directly (this hit me with a vengance in https://github.com/JuliaImages/ImageCore.jl/pull/52). Can something equivalent be done by testing the `indices`?

Reply:
Yeah I don't like these tests either. Testing the indices is more sane.